program/rust/src/processor.rs

```diff
@@ -84,5 +103,29 @@ pub fn process_instruction(
         DelProduct => del_product(program_id, accounts, instruction_data),
         UpdPermissions => upd_permissions(program_id, accounts, instruction_data),
         SetMaxLatency => set_max_latency(program_id, accounts, instruction_data),
+        InitPriceFeedIndex => init_price_feed_index(program_id, accounts, instruction_data),
+    }
+}
+
+fn reserve_new_price_feed_index(permissions_account: &AccountInfo) -> Result<u32, ProgramError> {
+    if permissions_account.data_len() < PermissionAccount::MIN_SIZE_WITH_LAST_FEED_INDEX {
+        let new_size = PermissionAccount::MIN_SIZE_WITH_LAST_FEED_INDEX;
+        let rent = Rent::get()?;
+        let new_minimum_balance = rent.minimum_balance(new_size);
+        pyth_assert(
+            permissions_account.lamports() >= new_minimum_balance,
+            ProgramError::AccountNotRentExempt,
+        )?;
+
+        permissions_account.realloc(new_size, true)?;
```

Comment:
nit: I think this could be false

Reply:
I think it's safer to leave `zero_init = true` because we're not really concerned about performance here.